### DISCUSSION TITLE
move mediacms installtion to urlpath

### DISCUSSION THREAD

**Discussion by lil0r** [mediacms-io/mediacms#581] [General]
Hi,

I like mediacms very much - great piece of work!
After some internal use, I am trying to publish my mediacms behind an nginx reverse proxy url.
For the url location of nginx, I am would like to move the mediacms path to a specific path.
Essential this is what I would like to achieve:
https://external.url/mediacms -> opnsense with nginx -> http://internal.ip.of.medicms/mediacms 

I tried initially with rewrite rules on nginx, there seem to be some hardcoded redirects to / that do not work here.

cheers

**Comment by mgogoulos** [comment]
Hey, not sure I understand the question, if you can elaborate more maybe I could give some hints

cheers

**Reply by lil0r** [reply]
My mediacms servers currently runs at http://172.16.188.10 within a DMZ. That DMZ is behind an opnsense with a nginx reverse proxy.
I have multiple webservers in the DMZ that are mapped via nginx reverse proxy (location + url rewrite), that are working fine.
So to access the different webservers in my DMZ from the internet, I am using https://example.com/webserverA, and  https://example.com/webserverB to route to those DMZ servers.

I tried to get https://example.com/mediacms routed to http://172.16.188.10/ via nginx. That is only working to some extend, actually the website shows, but no media plays back. since I am using URLrewritting on opnsense, I think this is breaking the media playback function of mediacms.
Now, if I can move the whole mediacms to http://172.16.188.10/mediacms directly, I could do simple URL mapping and was hoping that this would not break the media playback. I tried setting parameters in mediacms settings.py but then mediacms tells me "you are lost" ;)

Cheers

**Comment by lil0r** [comment]
@mgogoulos any pointers welcome :)

**Comment by mgogoulos** [comment]
You mention that this works to some extend (website loads but media are missing), can you check what the logs say on the browser console? You can see the locations/aliases on nginx on file deploy/local_install/mediacms.io - I'm sure you did ;) If the portal loads as you suggest, I'm assuming that /static files are served but not /media (that are the media files). Or it could be something else and not the case that media are not found after the new rules but some other issue, check if you see anything useful on the console and network tabs of the browser and share it here..

**Comment by lil0r** [comment]
Thanks @mgogoulos, console was a very good hint.
For testing purposes I have portmapping done from 443 to 8443, all the API calls and subsequent calls were all pointing to the base-url of mediacms ignoring the port and thus failing. I have moved the webserver to port 443 and media shows, but doesn't play back ;)
There are still countless links like view/featured/tags that would need custom rewrite rules - not to mention login and upload function. If I could just move the mediacms installation on the webserver itself to an https://webserver/mediacms url, I would only need one rewrite rule. That would be obviously much easier and elegant. Is this planned for mediacms at some stage?

cheers

**Comment by lil0r** [comment]
minor update, gave up on the urlpath configuration and changed to SNI based routing, meaning I use https://mediacms.domain.de instead of https://domain.de/mediacms to access. Fine for me at the moment, if I get time I raise a feature request.
I am troubleshooting some encoding issues with large files and post my findings separately.
